src/cryptenroll/cryptenroll.c

```diff
@@ -486,6 +499,28 @@ static int prepare_luks(
                                 strlen(envpw));
                 if (r < 0)
                         return log_error_errno(r, "Password from environment variable $PASSWORD did not work.");
+        } else if(arg_unlock_file) {
+                char *password;
+                size_t password_len;
+
+                r = crypt_keyfile_device_read(
+                                cd,
+                                arg_unlock_file,
+                                &password,
+                                &password_len,
+                                0, 0, 0);
+                if (r < 0)
+                        return log_error_errno(r, "Unlocking via keyfile failed: %m");
```

Comment:
this is not unlocking, it's reading, so as mentioned: `"Reading keyfile %s failed: %m", arg_unlock_file`